### PR TITLE
[FS] Check if transaction hit the mempool

### PR DESCRIPTION
**Pull request by fenix2222**

### CODE REVIEW

File: src/Stratis.SmartContracts.Tests.Common/MockChain/MockChainNode.cs [hunk 1]
```diff
@@ -177,8 +177,14 @@ public Result<WalletSendTransactionModel> SendTransaction(Script scriptPubKey, M
                 Sender = sender ?? this.MinerAddress.Address,
                 WalletName = this.WalletName
             };
-            JsonResult response = (JsonResult)this.smartContractsController.BuildAndSendCreateSmartContractTransaction(request);
-            return (BuildCreateContractTransactionResponse)response.Value;
+
+            IActionResult result = this.smartContractsController.BuildAndSendCreateSmartContractTransaction(request);
+            if (result is JsonResult response)
```
Comment (codingupastorm) [inline src/Stratis.SmartContracts.Tests.Common/MockChain/MockChainNode.cs]: Hmm what's this change for?

Comment (fenix2222) [reply src/Stratis.SmartContracts.Tests.Common/MockChain/MockChainNode.cs]: Test assumes that result will always be of type JsonResult, however, we are sending a BadRequest response which is of a different type but uses the same IAsyncResult interface.

Comment (codingupastorm) [reply src/Stratis.SmartContracts.Tests.Common/MockChain/MockChainNode.cs]: Ah okay, I think this is breaking the Sct tool too

Comment (codingupastorm) [reply src/Stratis.SmartContracts.Tests.Common/MockChain/MockChainNode.cs]: We need to look into this more, make sure that Sct and the wallet can handle errors / bad request. Atm I think it's expecting errors to be output in "Message" inside the json or something similar

Comment (fenix2222) [reply src/Stratis.SmartContracts.Tests.Common/MockChain/MockChainNode.cs]: Wallet is definitely fine as it handles bad requests from server. I will check SC tool

Comment (fenix2222) [reply src/Stratis.SmartContracts.Tests.Common/MockChain/MockChainNode.cs]: Sct also handles ErrorResponses: https://github.com/stratisproject/Stratis.SmartContracts.Tools.Sct/blob/master/Stratis.SmartContracts.Tools.Sct/Deployment/HttpContractDeployer.cs

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs [hunk 2]
```diff
@@ -329,9 +331,18 @@ public IActionResult BuildAndSendCreateSmartContractTransaction([FromBody] Build
                 return this.Json(response);
 
             Transaction transaction = this.network.CreateTransaction(response.Hex);
-            this.walletManager.ProcessTransaction(transaction, null, null, false);
+
             this.broadcasterManager.BroadcastTransactionAsync(transaction).GetAwaiter().GetResult();
 
+            // Check if transaction was actually added to a mempool.
+            TransactionBroadcastEntry transactionBroadCastEntry = this.broadcasterManager.GetTransaction(transaction.GetHash());
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs]: It looks like this code and the changes below are the same? Perhaps put it in a shared method?

Comment (fenix2222) [reply src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs]: I thought about it already, but there is not much benefit. Controllers don't share common base class and there wait it constructs a result would not improve code that much.

**Pull request merged**